phoenix-shared/src/commonMain/paymentsdb/fr.acinq.phoenix.db/migrations/2.sqm

```diff
@@ -0,0 +1,22 @@
+-- Migration: v2 -> v3
+--
+-- Changes:
+-- * Added table payments_metadata
+
+import fr.acinq.phoenix.db.payments.LNUrlBase;
+import fr.acinq.phoenix.db.payments.LNUrlMetadata;
+import fr.acinq.phoenix.db.payments.LNUrlSuccessAction;
+
+CREATE TABLE IF NOT EXISTS payments_metadata (
+    type INTEGER NOT NULL,
+    id TEXT NOT NULL,
+    lnurl_base_type TEXT AS LNUrlBase.TypeVersion,
+    lnurl_base_blob BLOB,
+    lnurl_description TEXT,
```

Comment:
What is the lnurl's description? Is it the user's comment that can be attached to the payment?

Reply:
You made me aware of this requirement the other day. There is a priority for deciding what the payment description is:

1. If the user has manually entered a (short) description (`payments_metdata.user_description`)
2. If there's a corresponding LNUrl.Pay, then it contains a description in the metadata (`payments_metadata.lnurl_description`)
3. The normal description extracted from the WalletPayment (most often via the invoice)

In the context of LNUrl.Pay, this is particularly important, because the lightning invoice uses a descriptionHash (as per the spec), which must match `SHA256(LNUrl.Pay.metadata.raw)`.

> Is it the user's comment [...] ?

I envision the possibility for the user to separately:
- override the (short) description
- add (long) notes

The ability to override the short description is nice. Because, as we witness on credit card statements, when companies enter descriptions they are often horrible. And it's these short descriptions that appear in most places throughout the app.

Thus, the payments_metadata table will ultimately have columns:
- user_description
- user_notes